.github/workflows/test.yml

```diff
@@ -26,182 +26,209 @@ jobs:
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE }}
+          aws-region: ${{ env.AWS_REGION }}
+      - name: Clean and delete stack
+        if: (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
+        working-directory: ./convert
+        run: |
+          ./convert.py clean-stack --stack-name $NAME --region $AWS_REGION --remove-security-group-references --delete
+          ./convert.py delete-stack --stack-name $NAME --region $AWS_REGION --delete
+      - name: Upload diagnostic data
+        if: always() && (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
+        uses: actions/upload-artifact@v2
+        with:
+          name: Diagnostic Data
+          path: ./
+          retention-days: 14
 
-    - name: Authenticate with AWS
-      uses: aws-actions/configure-aws-credentials@v2
-      if: always() && (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
-      with:
-        role-to-assume: ${{ secrets.AWS_IAM_ROLE }}
-        aws-region: ${{ env.AWS_REGION }}
-    - name: Delete stack w/CDK
-      if: always() && (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
-      working-directory: ./cdk
-      run: |
-        cdk destroy --force
-    - name: Destroy Infrastructure
-      if: always() && (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
-      working-directory: ./convert/terraform
-      run: |
-        terraform destroy -auto-approve
-    - name: Destroy Infrastructure
-      if: always() && (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
-      working-directory: ./convert/cloudformation-only
-      run: |
-        terraform destroy -auto-approve
-    - name: Fail without deploy
-      if: ${{ github.event.pull_request.draft == false && ! (contains(github.event.pull_request.labels.*.name, 'deploy-test') || contains(github.event.pull_request.labels.*.name, 'no-deploy-needed') || github.ref == 'refs/heads/master') }}
-      run: |
-        echo "Deploy tests required on non-draft PRs. Please add 'deploy-test' label".
-        exit 1
+      - name: Authenticate with AWS
+        uses: aws-actions/configure-aws-credentials@v2
+        if: always() && (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
+        with:
+          role-to-assume: ${{ secrets.AWS_IAM_ROLE }}
+          aws-region: ${{ env.AWS_REGION }}
+      - name: Delete stack w/CDK
+        if: always() && (contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master')
+        env:
+          JSII_SILENCE_WARNING_DEPRECATED_NODE_VERSION: 1
+        working-directory: ./cdk
+        run: |
+          MAX_RETRIES=3
```

Comment:
I suspect this retry was added trying to troubleshoot problems with deletions, but also probably wasn't/won't be the actual solution. I'd rather not do a retry. If it fails, it fails.

Reply:
There are instances where the delete fail due to a sg, but I was able to delete it manually(hints at timing), given this is a delete, Id say its worth a retry rather than ~1hr for a re-run.

Reply:
removed